### DISCUSSION TITLE
Invalid Schema renders correctly

### DISCUSSION THREAD

**Discussion by guizoxxv** [OAI/OpenAPI-Specification#4105] [Q&A]
Hello,

There are two things that I would like to have in OpenAPI that the seems not to be possible according to the documentation. Strangely, if I try to implement them, the result HTML is rendered as I would expect, even though schema validation errors also appear.

1. Extend/override properties of a parameter defined externally.

Example:
```yaml
# ...

/external-param:
  get:
    description: Uses a parameter defined in another file.
    parameters:
      - $ref: reusableParam.yaml
    responses:
      200:
        description: OK
/external-param-combined:
  get:
    description: Uses a parameter defined in another file and combines with local parameters.
    parameters:
      - name: reusableParam
        allOf:
          - $ref: reusableParam.yaml
          - required: true
          - schema:
            type: number
            default: 50
    responses:
      200:
        description: OK

# ...
```

2. Extend/Override a set of responses defined externally.

Example:
```yaml
# ...

    /external-responses-combined:
      get:
        description: Uses multiple responses defined in another file and combines with local responses.
        responses:
          allOf:
            - $ref: '#/components/responses/GeneralResponses'
            - 200:
                description: OK

components:
  responses:
    GeneralResponses:
      $ref: generalResponses.yaml

# ...
```

Also strange, when executing the same code in a local [swagger-ui](https://github.com/swagger-api/swagger-ui) project (`npm run dev`) OR with a local static files server (I used [Caddy](https://caddyserver.com/) and [http-server](https://github.com/http-party/http-server)) it works and doesn't show any schema validation errors.

Can someone help me understand this behaviour? Here is a complete example:
repo: https://github.com/guizoxxv/open-api-example
html: https://guizoxxv.github.io/open-api-example/

Thank you.

**Comment by SensibleWood** [comment]
@guizoxxv there's a few things to unpack in here:

* Parameters and Responses Objects are _not_ defined as Schema Objects so using the construct you are using you'll see inconsistent results, depending on how "loose" the tooling maker is in interpreting the `allOf` approach.
* Your validation error on GitHub Pages is down to the fact that you are loading from a URL in `index.html` but all your references are local, which is why Swagger UI barfs when deployed.
* That barfing looks like it has persuaded Swagger UI that there is indeed an issue with the Responses object, and it's given up with a misleading error.

This leads back to one key point: An OpenAPI is not a feature complete JSON Schema document, so you cannot use all the features of JSON Schema throughout. You'll get inconsistent results. I think, but am not sure, that one day OpenAPI will also be a complete JSON Schema document, but it isn't at 3.0.1.

**Comment by handrews** [comment]
I'm assuming from the lack of response that the answer was accepted as it seems pretty thorough.  Closing.
